src/main/java/org/opentripplanner/middleware/controllers/api/ApiUserController.java

```diff
@@ -84,20 +94,31 @@ private boolean userHasKey(ApiUser user, String apiKeyId) {
                 .anyMatch(apiKey -> apiKeyId.equals(apiKey.keyId));
     }
 
+    /**
+     * Authenticate user with Auth0 based on username (email) and password. If successful, return the complete Auth0
+     * token else null.
+     */
+    private TokenHolder authenticateAuth0User(Request req, Response res) {
```

Comment:
Before this method gets called, `Auth0Connection.isAuthHeaderPresent()` will be called and check for an auth0 authorization header. The 3rd party user does not have an Auth0 bearer token to provide at this point, so this method will never execute. As a result, an exception in `spark.before(API_PREFIX + "/secure/*"` in `OtpMiddlewareMain` needs to be added. The corresponding test `canSimulateApiUserFlow()` also needs to use a mock call without bearer token.

Reply:
Chicken/Egg. Good spot. Have updated.